internal/pkg/cli/local_run.go

```diff
@@ -56,15 +119,182 @@ func (o *localRunOpts) Validate() error {
 
 // Ask prompts the user for any unprovided required fields and validates them.
 func (o *localRunOpts) Ask() error {
-	//TODO(varun359): Validate and Ask SvcEnvName
-
+	if err := o.validateOrAskWorkloadName(); err != nil {
+		return err
+	}
+	if err := o.validateOrAskEnvName(); err != nil {
+		return err
+	}
 	return nil
 }
 
 // Execute builds and runs the workload images locally.
 func (o *localRunOpts) Execute() error {
-	//TODO(varun359): Get build information from the manifest and task definition for workloads
+	taskDef, err := o.ecsLocalClient.TaskDefinition(o.appName, o.envName, o.wkldName)
+	if err != nil {
+		return fmt.Errorf("get task definition: %w", err)
+	}
+
+	_, err = o.getDecryptedSSMSecrets(taskDef, o.sess)
+	if err != nil {
+		return err
+	}
+
+	_, err = o.getDecryptedSecretManagerSecrets(taskDef, o.sess)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (o *localRunOpts) getDecryptedSSMSecrets(taskDef *awsecs.TaskDefinition, awsSession *session.Session) ([]decryptedSecret, error) {
+	var ssmSecrets []decryptedSecret
+	secrets := taskDef.Secrets()
+	for _, secret := range secrets {
+		if !strings.HasPrefix(secret.ValueFrom, "arn:aws:secretsmanager:") {
```

Comment:
did we mean `ssm` here 👀 ?

Reply:
Let me know what you think of this!
```go
// In internal/aws/secretsmanager/secretsmanager.go
const namespace = "secretsmanager"

// IsService returns true if the given ARN is a secretsmanager ARN.
func IsService(arn string) bool  // Use arn.Parse Ref: https://docs.aws.amazon.com/sdk-for-go/api/aws/arn/#Parse to check whether the arn is of secretsmanager. 


// In internal/aws/ssm/ssm.go
const namespace = "ssm"

// IsService returns true if the given ARN is an ssm ARN.
func IsService(arn string) bool
```

This lets individual `aws/x` package manages the service namespace that they should be looking for; using `arn.Parse` makes the code more robust - checking prefix only works for the "aws" partition, but there are also china partition (looks like "arn:aws-cn:ssm") and govcloud partition ("arn:aws-gov:ssm")